plugins/router/template/router.go

```diff
@@ -144,6 +145,20 @@ func newTemplateRouter(cfg templateRouterCfg) (*templateRouter, error) {
 	return router, nil
 }
 
+func endpointsForAlias(alias ServiceAliasConfig, svc ServiceUnit) []Endpoint {
+	if len(alias.PreferPort) == 0 {
+		return svc.EndpointTable
+	}
+	endpoints := make([]Endpoint, 0, len(svc.EndpointTable))
+	for i := range svc.EndpointTable {
+		endpoint := svc.EndpointTable[i]
+		if endpoint.PortName == alias.PreferPort || endpoint.Port == alias.PreferPort {
```

Comment:
I think the concern is that the different ports mean something different.
Why would I have both 80 and 443 exposed on a service?  I might have 8080
and 8081 that I want round robin-ed, but I can't think of a real reason to
do that.  There was no scenario I could construct for a route where we
wanted to expose 1 port to multiple backing endpoint ports.  Instead, all
the scenarios were where we wanted to have 80+443, then maybe expose
2800-2900 for additional extension.  But even then, it's not clear that
extra port is that useful.

On Tue, Sep 15, 2015 at 2:34 PM, Ram Ranganathan <notifications@github.com>
wrote:

> In plugins/router/template/router.go
> <https://github.com/openshift/origin/pull/4606#discussion_r39547794>:
>
> > @@ -144,6 +145,20 @@ func newTemplateRouter(cfg templateRouterCfg) (*templateRouter, error) {
> >  	return router, nil
> >  }
> >
> > +func endpointsForAlias(alias ServiceAliasConfig, svc ServiceUnit) []Endpoint {
> > +	if len(alias.PreferPort) == 0 {
> > +		return svc.EndpointTable
> > +	}
> > +	endpoints := make([]Endpoint, 0, len(svc.EndpointTable))
> > +	for i := range svc.EndpointTable {
> > +		endpoint := svc.EndpointTable[i]
> > +		if endpoint.PortName == alias.PreferPort || endpoint.Port == alias.PreferPort {
>
> Catching up this week. Should be easy to add multiple ports in here, no?
> Let the ServiceAliasConfig have a list of preferred ports and
> add 'em to a map[string]bool mapR here and then later just change this
> check:
> endpoint.PortName == alias.PreferPort || endpoint.Port == alias.PreferPort
> to mapR[endpoint.PortName] || mapR[endpoint.Port] or something to that
> effect.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/4606/files#r39547794>.
>



-- 
Clayton Coleman | Lead Engineer, OpenShift